app/models/lti_deployment.rb

```diff
@@ -2,4 +2,29 @@ class LtiDeployment < ApplicationRecord
   belongs_to :course, optional: true
   belongs_to :lti_client
   validates :external_deployment_id, uniqueness: { scope: :lti_client }
+
+  def get_students
+    auth_data = lti_client.get_oauth_token(['https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly'])
+    names_service = LtiService.find_by!(lti_deployment: self, service_type: 'namesroles')
```

Comment:
I generally prefer using explicit associations on queries like these, e.g. `self.lti_services.find(...)` (which will raise an error if no record is found)